adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -539,4 +538,15 @@ private void doUploadMessage(final RoutingContext ctx, final String tenant, fina
             });
         }
     }
+
+    private static Integer getQoSLevel(String qosValue) {
+        try {
+            if (qosValue == null)
```

Comment:
check for null not necessary here, it will be transfered to a `NumberFormatException` by `parseInt`.

Reply:
Sure, but this handles the case when the qos header value is not set by the client. That should be differentiated from an invalid qos header value

Reply:
Ok, I see :+1: